### PR TITLE
Rename package deviceplugin => devicemanager.

### LINKED ISSUE #58795: Rename package deviceplugin to devicemanager for consistency.
/kind cleanup
/sig node

### PR DESCRIPTION
**Pull request by ConnorDoyle**
**What this PR does / why we need it**:

Fixes #58795
/kind cleanup

Rename package `deviceplugin` to `devicemanager` for consistency.

We already have components named Container manager and CPU manager. The device plugin package similarly contains an interface called `Manager`. The fact that the manager has plugins is somewhat incidental to the purpose of the package itself.

Note that this rename only affects internal API. The external gRPC interface still exports a package called deviceplugin.

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by ConnorDoyle** [comment]
cc @jiayingz @RenaudWasTaken

**Comment by tengqm** [comment]
@ConnorDoyle Can we hold this a little bit? There are some non-trivial PRs under review (e.g. #52616, #56398). This rename is gonna make the rebase pretty difficult a job.

File: pkg/kubelet/cm/devicemanager/kubelet_internal_checkpoint
Comment (RenaudWasTaken) [inline pkg/kubelet/cm/devicemanager/kubelet_internal_checkpoint]: We really need to fix the internal checkpoint not being cleaned up bug...

**Review by RenaudWasTaken — CHANGES_REQUESTED** [review]
Remove the kubelet checkpoint file :)

LGTM though I'd like us to focus on getting the 5 other pending refactor PRs in.

**Comment by ConnorDoyle** [comment]
/hold
Wait until some of the long-standing PRs in the deviceplugin are merged.

**Comment by ConnorDoyle** [comment]
@RenaudWasTaken thanks, removed the checkpoint file.

**Comment by ConnorDoyle** [comment]
attn @vishh @jiayingz

**Comment by RenaudWasTaken** [comment]
still /lgtm to me :)

File: pkg/kubelet/cm/devicemanager/kubelet_internal_checkpoint
Comment (jiayingz) [reply pkg/kubelet/cm/devicemanager/kubelet_internal_checkpoint]: remove this file?

**Review by jiayingz — COMMENTED** [review]
/lgtm

File: pkg/kubelet/cm/devicemanager/kubelet_internal_checkpoint
Comment (ConnorDoyle) [reply pkg/kubelet/cm/devicemanager/kubelet_internal_checkpoint]: Removed!

**Comment by ConnorDoyle** [comment]
@vikaschoudhary16 @jiayingz @RenaudWasTaken could I get another lgtm when tests go green? Removed the checkpoint file I committed by accident.

**Pull request merged**
